geowave/src/main/scala/geotrellis/spark/io/geowave/GeowaveLayerWriter.scala

```diff
@@ -15,122 +17,199 @@ import com.typesafe.scalalogging.slf4j._
+        for (
+          statsAggregator <- managed(new StatsCompositionTool(new DataStoreStatisticsProvider(
+            adapter,
+            index,
+            true),
+            GeoWaveStoreFinder.createDataStatisticsStore(configOptions)))
+        ) {
+          val kvGen = new AccumuloKeyValuePairGenerator[GridCoverage](
+            adapter,
+            index,
+            statsAggregator,
+            DataStoreUtils.UNCONSTRAINED_VISIBILITY.asInstanceOf[VisibilityWriter[GridCoverage]]);
+
+          adapter.convertToIndex(index, image).foreach { x => (accumuloKvs += (kvGen.constructKeyValuePairs(adapter.getAdapterId.getBytes, x).toList.map { kv => (kv.getKey, kv.getValue) })) }
+        }
       }
-    })
+      accumuloKvs.foldLeft(Iterable[(Key, Value)]())(_ ++ _)
+    }
+    val index = (new SpatialDimensionalityTypeProvider.SpatialIndexBuilder).createIndex();
+    val indexName = index.getId.getString;
+    val tableName = AccumuloUtils.getQualifiedTableName(as.geowaveNamespace, indexName);
+
+    val gwMetadata = new java.util.HashMap[String, String](); gwMetadata.put("cellType", cellType)
+    val basicOperations = new BasicAccumuloOperations(
+      as.zookeeper,
+      as.accumuloInstance,
+      as.accumuloUser,
+      as.accumuloPass,
+      as.geowaveNamespace)
+    val adapter = new RasterDataAdapter(coverageName, gwMetadata, image, 256, true, false, Array.fill[Array[Double]](image.getNumSampleDimensions)(Array(0.0)))
+    //make sure adapter and index get written       
+    val adapterStore = new AccumuloAdapterStore(basicOperations)
+    adapterStore.addAdapter(adapter)
+    val indexStore = new AccumuloIndexStore(basicOperations)
+    indexStore.addIndex(index)
+    //make sure adapter and index are associated together in the mapping store
+    val mappingStore = new AccumuloAdapterIndexMappingStore(basicOperations)
+    mappingStore.addAdapterIndexMapping(new AdapterToIndexMapping(adapter.getAdapterId, Array(index.getId)))
+    AccumuloUtils.attachRowMergingIterators(
+      adapter,
+      basicOperations,
+      new AccumuloOptions,
+      index.getIndexStrategy().getNaturalSplits(),
+      indexName);
+    accumuloWriter.write(
+      rdd
+        .sortBy({ case (k, v) => SpatialKey.keyToTup(k.asInstanceOf[SpatialKey]) })
+        .groupBy({ case (k, _) => k.asInstanceOf[SpatialKey]._1 })
+        .map(_._2)
+        .mapPartitions(partitions => partitions.map(geotrellisKvToGeoWaveKv)).flatMap(i => i), AccumuloInstance(as.accumuloInstance, as.zookeeper, as.accumuloUser, new PasswordToken(as.accumuloPass)), tableName)
+
+    //compact and detach iterator
+    val conn = ConnectorPool.getInstance.getConnector(as.zookeeper,
+      as.accumuloInstance,
+      as.accumuloUser,
+      as.accumuloPass)
+    val ops = conn.tableOperations()
+    ops.compact(tableName, null, null, true, true)
```

Comment:
on looking it over again, it probably makes sense to compact the metadata table as well...stats are written much the same as tiles, there's an iterator responsible for merging them together, there are far fewer stats so it isn't of particular importance but `op.compact(AccumuloUtils.getQualifiedTableName(as.geowaveNamespace, AbstractGeowavePersistence.METADATA_TABLE), null, null, true, true))` probably makes sense here as well